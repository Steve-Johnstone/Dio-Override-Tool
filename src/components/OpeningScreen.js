import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import axios from 'axios';
import Header from './Header';
import 'bulma/css/bulma.min.css';

const OpeningScreen = ({
	url,
	setUrl,
	setOverrideList,
	setSelectedOverrides,
}) => {
	const [overrides, setOverrides] = useState([]);
	const [error, setError] = useState('');

	useEffect(() => {
		const axiosPosts = async () => {
			const response = await axios({
				method: 'get',
				url: 'http://localhost:8080/mock/override-data.json',
			});
			setOverrides(response.data.overrides);
		};
		axiosPosts();
	}, []);

	const handleClick = (e) => {
		e.preventDefault();

		if (overrides.length === 0) {
			return setError('ERROR: no connection to the host server.');
		}
		setOverrideListIntoState();
	};

	const setOverrideListIntoState = () => {
		if (url.includes('?')) {
			let [pagePath, overrideSection] = url.split('?');
			let preSelectedOverrides = overrideSection.substring(9).split(',');

			if (!overrides[pagePath]) {
				return setError('ERROR: no overrides found for provided URL.');
			}
			setSelectedOverrides(preSelectedOverrides);
			setUrl(pagePath);
			setOverrideList(overrides[pagePath]);
		} else {
			if (!overrides[url]) {
				return setError('ERROR: no overrides found for provided URL.');
			}
			setOverrideList(overrides[url]);
		}
	};

	return (
		<div className='main-page-container'>
			<div className='header level-item is-justify-content-space-between'>
				<Header />
			</div>

			<div className='container p-4'>
				<div>
					<form className='field py-2'>
						<div style={{ width: '40%' }} className='control'>
							<label htmlFor='text' className='label'>
								Local URL of the page
							</label>
							<div>
								<input
									name='url'
									className='input is-link level-item'
									placeholder='www.url.com'
									type='text'
									onChange={(e) => setUrl(e.target.value)}
								></input>
							</div>
							<button
								onClick={(e) => handleClick(e)}
								className='button mt-4'
								name='Go'
								id='go-button'
							>
								<Link style={{ color: 'white' }} to='/overrides'>
									GO
								</Link>
							</button>
							<div className={error ? 'message is-danger mt-5' : ''}>
								<div className={error ? 'message-body' : ''}>{error}</div>
							</div>
						</div>
					</form>
				</div>
			</div>
		</div>
	);
};

OpeningScreen.propTypes = {
	url: PropTypes.string,
	setUrl: PropTypes.func,
	setOverrideList: PropTypes.func,
	setSelectedOverrides: PropTypes.func,
};

export default OpeningScreen;
