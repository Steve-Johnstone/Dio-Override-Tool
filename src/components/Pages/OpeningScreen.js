import { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';
import Header from '../Layout/Header';
import Button from '../UI/Button';
import ErrorDisplay from '../UI/ErrorDisplay';
import axios from 'axios';
import 'bulma/css/bulma.min.css';

const OpeningScreen = ({ setUrl, setOverrideList, setSelectedOverrides }) => {
	const [overrides, setOverrides] = useState([]);
	const [error, setError] = useState('');
	const [userInput, setUserInput] = useState('');
	const navigate = useNavigate();

	useEffect(() => {
		const axiosPosts = async () => {
			const response = await axios({
				method: 'get',
				url: 'http://localhost:8080/mock/override-data.json',
			});
			setOverrides(response.data.overrides);
		};
		axiosPosts();
	}, []);

	const handleClick = (e) => {
		e.preventDefault();

		if (overrides.length === 0) {
			setError('ERROR: no connection to the host server.');
			return;
		}
		setOverrideListIntoState();
	};

	const handlePaste = (input) => {
		setOverrideListIntoState(input);
	};

	const setOverrideListIntoState = (pastedInput) => {
		let url = pastedInput ? pastedInput : userInput;

		if (url.includes('?')) {
			let [pagePath, overrideSection] = url.split('?');
			let preSelectedOverrides = overrideSection.substring(9).split(',');

			if (!overrides[pagePath]) {
				setError('ERROR: no overrides found for provided URL.');
				return;
			}
			setSelectedOverrides(preSelectedOverrides);
			setUrl(pagePath);
			setOverrideList(overrides[pagePath]);
		} else {
			if (!overrides[url]) {
				setError('ERROR: no overrides found for provided URL.');
				return;
			}
			setUrl(url);
			setOverrideList(overrides[url]);
		}
		navigate('/overrides');
	};

	return (
		<div className='main-page-container'>
			<div className='header-container level-item is-justify-content-space-between'>
				<Header />
			</div>

			<div className='container p-4'>
				<div>
					<form className='field py-2'>
						<div style={{ width: '40%' }} className='control'>
							<label htmlFor='text' className='label'>
								Local URL of the page
							</label>
							<div>
								<input
									name='url'
									className='input is-link level-item'
									placeholder='www.hotels.com'
									type='text'
									onChange={(e) => setUserInput(e.target.value)}
									onPaste={(e) => handlePaste(e.clipboardData.getData('Text'))}
								></input>
							</div>
							<Button
								className='is-info mt-4'
								name='Go'
								text='GO'
								onClick={(e) => handleClick(e)}
							></Button>

							{error && <ErrorDisplay error={error} />}
						</div>
					</form>
				</div>
			</div>
		</div>
	);
};

OpeningScreen.propTypes = {
	setUrl: PropTypes.func,
	setOverrideList: PropTypes.func,
	setSelectedOverrides: PropTypes.func,
};

export default OpeningScreen;
