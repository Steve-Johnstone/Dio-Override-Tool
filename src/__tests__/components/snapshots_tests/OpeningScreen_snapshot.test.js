import renderer from 'react-test-renderer';
import OpeningScreen from '../../../components/OpeningScreen';
import { MemoryRouter } from 'react-router-dom';

describe('OpeningScreen component', () => {
	const { setUrl, setOverrideList, setSelectedOverrides } = jest.fn();

	it('should display the DOM correctly', () => {
		const tree = renderer
			.create(
				<MemoryRouter>
					<OpeningScreen
						setUrl={setUrl}
						setOverrideList={setOverrideList}
						setSelectedOverrides={setSelectedOverrides}
					/>
				</MemoryRouter>
			)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
