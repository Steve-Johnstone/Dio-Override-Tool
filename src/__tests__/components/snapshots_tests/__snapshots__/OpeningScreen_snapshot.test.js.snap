// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningScreen component should display the DOM correctly 1`] = `
<div
  className="main-page-container"
>
  <div
    className="header level-item is-justify-content-space-between"
  >
    <div
      className="is-flex is-justify-content-space-between"
    >
      <img
        alt="hcom logo"
        className="logo"
        data-testid="logo"
        src="image/hotelscom-logo.png"
      />
      <p>
        Dio Override Tool
      </p>
    </div>
  </div>
  <div
    className="container p-4"
  >
    <div>
      <form
        className="field py-2"
      >
        <div
          className="control"
          style={
            Object {
              "width": "40%",
            }
          }
        >
          <label
            className="label"
            htmlFor="text"
          >
            Local URL of the page
          </label>
          <div>
            <input
              className="input is-link level-item"
              name="url"
              onChange={[Function]}
              placeholder="www.hotels.com"
              type="text"
            />
          </div>
          <div>
            <button
              className="button is-info mt-4"
              name="Go"
              onClick={[Function]}
            >
              <a
                href="/overrides"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                GO
              </a>
            </button>
          </div>
          
        </div>
      </form>
    </div>
  </div>
</div>
`;
